### ISSUE TITLE
Feature: Schema editing support for localSchemaFile 

### ISSUE THREAD

**Issue by Cellule** [apollographql/vscode-graphql#253]
Originally from: https://github.com/apollographql/vscode-graphql/issues/247

When working with localSchemaFile it would be nice to have minimal schema editing features.

Here's a few ideas that comes to mind
- "Go to Definition" when hovering on a field's type
- Basic validation like showing an error when the type of a field is unknown
- Autocomplete when typing a field's type

As mentioned in the original issue, maybe having a rover config would help, but I'd still love some support considering the extension has all the necessary schema information to provide an improved developer experience
https://www.apollographql.com/docs/graphos/schema-design/connectors/vs-code

For reference in https://marketplace.visualstudio.com/items?itemName=GraphQL.vscode-graphql
I get type validation, type information on hover and go to definition

<img width="1471" alt="Screenshot 2024-12-17 at 9 40 56 AM" src="https://github.com/user-attachments/assets/1cd9bbdf-2eb2-4103-b3dc-2c6f5443b1a6" />

<img width="488" alt="Screenshot 2024-12-17 at 9 41 55 AM" src="https://github.com/user-attachments/assets/57c692e7-f111-4157-a110-8773790627b9" />

**Comment by dylanwulf** [comment]
I agree with this, I would at least like to have "Go to definition" working on a local schema file without requiring rover
